Delete bill

patrick55529617 opened the pull request:
Merge Delete-bill:

1. Add 'multiple delete' function.
2. Add logout function.
3. Add a button in '/test_show_all/' to link the '/add_billing/ page.
4. Improve the UI of page '/test_show_all/'.
5. Fix the timezone to Asia/Taipei

patrick55529617 committed:
Use button to delete ver1.

patrick55529617 committed:
1. Add 'multiple delete' function.
2. Add logout function.
3. Add a button in '/test_show_all/' to link the '/add_billing/ page.
4. Improve the UI of page '/test_show_all/'.

patrick55529617 committed:
1. Add 'multiple delete' function.
2. Add logout function.
3. Add a button in '/test_show_all/' to link the '/add_billing/ page.
4. Improve the UI of page '/test_show_all/'.

renaisho commented on BillingApp/user/views.py:
```diff
@@ -1,6 +1,11 @@
 class LoginView(AuthLoginView):
     template_name = 'registration/login.html'
+
+def Logout(request):
```
maybe we could use [`LogoutView`](https://docs.djangoproject.com/en/3.0/topics/auth/default/#django.contrib.auth.views.LogoutView)

renaisho commented on BillingApp/BillingInfo/views.py:
```diff
@@ -23,9 +23,11 @@ def post(self, request):
+    for deleted_index in check_box_list:
+        deleted_bill = BillInfo.objects.get(id=deleted_index)
+        deleted_bill.delete()
     return HttpResponseRedirect("/test_show_all/")
```
we should use named url instead of hard code url, see this: [`reverse`](https://docs.djangoproject.com/en/3.0/ref/urlresolvers/#reverse)

renaisho commented on BillingApp/BillingInfo/templates/test_show_all.html:
```diff
@@ -2,41 +2,68 @@
 <head>
     <meta charset="UTF-8">
     <title>TEST</title>
+    <style>
```
we could extract this to a css file in `static/..`, see [this](https://docs.djangoproject.com/en/3.0/intro/tutorial06/)

renaisho commented on BillingApp/BillingApp/settings.py:
```diff
@@ -46,7 +46,7 @@
     {
         'BACKEND': 'django.template.backends.django.DjangoTemplates',
-        'DIRS': [],
+        'DIRS': [os.path.join(BASE_DIR, 'templates')],
```
why we need to add this?
Actually we could access `app/templates` if APP_DIRS is True

patrick55529617 committed:
1. Split base.html to make html file cleaner.
2. Add the calendar selector.

patrick55529617 committed:
Merge branch 'delete_bill' of https://github.com/patrick55529617/BillingApp into delete_bill

patrick55529617 committed:
Fix 1, 2, 4 PR comments.

patrick55529617 committed:
1. Merge master branch and fix conflicts.
2. Add a button link to registration page from login page.

commit c10a8f0d7a9a1de6981eb312c73c2f308109b951
Merge: 78ea0a2 eef5c4a
Author: patrick55529617 <patrick55529617@gmail.com>
Date:   Mon Apr 6 22:50:55 2020 +0800

    Merge pull request #6 from patrick55529617/registration

    add registration page

commit eef5c4a5047c50dc27e402ba76ff801a650af4d0
Author: Ho Renais <renais@foundi.info>
Date:   Sat Apr 4 17:57:58 2020 +0800

    add registration page

patrick55529617 committed:
Merge branch 'master' into delete_bill

patrick55529617 committed:
Add static/css/main.css to manage css setting.

renaisho commented on BillingApp/BillingInfo/templates/base.html:
```diff
@@ -0,0 +1,54 @@
+    <link rel="stylesheet" href="//code.jquery.com/ui/1.12.1/themes/base/jquery-ui.css">
+    <script src="https://code.jquery.com/ui/1.12.1/jquery-ui.js"></script>
+    <script type="text/javascript" onclick="document.getElementById('p').style.display=''">
+    $(document).ready(function(){
```
this should not be in base.html
I think not all html will use this script

renaisho commented on BillingApp/BillingInfo/templates/base.html:
```diff
@@ -0,0 +1,54 @@
+          $( "#datepicker" ).datepicker();
+        } );
+    </script>
+    <style>
```
same above

renaisho commented on BillingApp/BillingInfo/templates/test_show_all.html:
```diff
@@ -1,53 +1,28 @@
             <th>刪除</th>
             </tr>
-    <form action="./del/" method="POST"> {% csrf_token %}
+    <form action="/del/" method="POST"> {% csrf_token %}
```
use named url
like `{% url 'del' %}`, see [this](https://docs.djangoproject.com/en/3.0/topics/http/urls/#reversing-namespaced-urls)

renaisho commented on BillingApp/BillingInfo/templates/test_show_all.html:
```diff
@@ -1,53 +1,28 @@
     <br>
+    <center>
+        <form action="/query_one/" method="POST"> {% csrf_token %}
+        <div style="display:none" id="p1">請輸入日期：</p><p>Date: <input type="text" id="datepicker" name="one_date"><input type="submit" value="查詢"></p>
```
`<p>` is missing

renaisho commented on BillingApp/user/views.py:
```diff
@@ -1,11 +1,12 @@
-    return HttpResponseRedirect('/user/login')
\ No newline at end of file
+class LogoutView(AuthLogoutView):
+    next_page = settings.LOGIN_URL
```
cool

renaisho commented on BillingApp/BillingInfo/templates/query_one.html:
```diff
@@ -0,0 +1,44 @@
+{% extends 'base.html' %}
```
I think we could just use the template `test_show_all` with giving different data

renaisho commented on BillingApp/BillingInfo/templates/test_show_all.html:
```diff
@@ -1,31 +1,73 @@
+    <!-- Bootstrap core JavaScript -->
+    <script type="text/JavaScript" src="https://ajax.googleapis.com/ajax/libs/jquery/3.2.1/jquery.min.js"></script>
+    <script src="https://code.jquery.com/ui/1.12.1/jquery-ui.js"></script>
+    <script type="text/javascript" onclick="document.getElementById('p').style.display=''">
```
- `onclick` on `<script>` is not correct, we could remove this
- move this script to body
- use id instead of using tag directly to avoid bugs when next time we add `<button>` tag

renaisho commented on BillingApp/BillingApp/settings.py:
```diff
@@ -46,7 +46,6 @@
 TEMPLATES = [
     {
         'BACKEND': 'django.template.backends.django.DjangoTemplates',
-        'DIRS': [],
```
we can remain this field as `[]`

renaisho commented on BillingApp/user/templates/registration/login.html:
```diff
@@ -19,6 +19,7 @@
 
 		<input type="submit" value="login">
 		<input type="hidden" name="next" value="{{ next }}">
+		<input type="button" class='button' value="register" onclick="location.href='/user/registration/'" >
```
user `{% url 'user:registration' %}` instead

patrick55529617 committed:
1. Move main_css to billing_page.css
2. Delete query_one
